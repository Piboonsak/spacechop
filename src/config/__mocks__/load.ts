import fs from 'fs';
import yaml from 'js-yaml';
import path from 'path';

export const loadConfig = (filepath) => yaml.safeLoad(fs.readFileSync(filepath));

export default () => {
  const filepath = path.join(__dirname, '../__tests__/assets/valid.yml');
  if (fs.existsSync(filepath)) {
    try {
      const config = loadConfig(filepath);
      if (config) {
        return config;
      } else {
        throw new Error(`Configuration is empty, ${filepath}`);
      }
    } catch (err) {
      throw new Error(`Could not load ${filepath}:\n\n${err}`);
    }
  } else {
    throw new Error(`Could not find ${filepath}`);
  }
};
