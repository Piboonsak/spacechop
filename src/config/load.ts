import fs from 'fs';
import yaml from 'js-yaml';
import console from '../lib/console';

export const loadConfig = (filepath) => yaml.safeLoad(fs.readFileSync(filepath));

export default (filepath) => {
  if (fs.existsSync(filepath)) {
    try {
      return loadConfig(filepath) || null;
    } catch (err) {
      console.error(`Could not load ${filepath}`);
      console.error(err);
    }
  } else {
    console.error(`Could not find ${filepath}`);
  }
  return null;
};
