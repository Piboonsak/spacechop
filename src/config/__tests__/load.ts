import path from 'path';
import load from '../load';

jest.mock('../../lib/console');

describe('should load config.yml', () => {
  describe('invalid config', () => {
    it('should throw on missing file', () => {
      const filename = path.join(__dirname, 'assets/config.yml');
      expect(() => load(filename)).toThrowErrorMatchingSnapshot();
    });

    it('should throw on empty', () => {
      const filename = path.join(__dirname, 'assets/empty.yml');
      expect(() => load(filename)).toThrowErrorMatchingSnapshot();
    });

    it('should return null on bad yaml', () => {
      const filename = path.join(__dirname, 'assets/bad.yml');
      expect(() => load(filename)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('valid config', () => {
    it('should return config', () => {
      const filename = path.join(__dirname, 'assets/valid.yml');
      expect(load(filename)).toMatchSnapshot();
    });
  });
});
