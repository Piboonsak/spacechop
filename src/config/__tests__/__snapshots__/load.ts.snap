// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load config.yml valid config should return config 1`] = `
Object {
  "path": "/images/:preset/:image",
  "presets": Object {
    "t_720": Object {
      "steps": Array [
        Object {
          "$crop": null,
          "gravity": "center",
          "width": 720,
        },
        Object {
          "$format": Object {
            "type": "jpg",
          },
        },
        Object {
          "$compress": null,
          "lossy": true,
          "quality": 92,
        },
      ],
    },
  },
  "sources": Array [
    Object {
      "access_key_id": "access_key_id",
      "bucket_name": "bucket_name",
      "path": null,
      "region": "region",
      "s3": null,
      "secret_access_key": "secret_access_key",
    },
    Object {
      "http": null,
      "root": "https://example.com/images/:image",
    },
  ],
}
`;
